lib/parlour/types.rb

```diff
@@ -196,6 +196,43 @@ def describe
       end
     end
 
+    # A user-defined generic class with an arbitrary number of type
+    # parameters. This class assumes at least one type_param is
+    # provided, otherwise output will have empty type param lists.
+    class Generic < Type
+      sig { params(type: TypeLike, type_params: T::Array[TypeLike]).void }
+      def initialize(type, type_params)
+        @type = to_type(type)
+        @type_params = type_params.map { |p| to_type(p) }
+      end
+
+      sig { params(other: Object).returns(T::Boolean) }
+      def ==(other)
+        self.class === other && type_params == other.type_params
```

Comment:
Maybe check that `type == other.type` here too

Reply:
:+1: Yep, that seems necessary haha